python/lsst/analysis/tools/actions/plot/barPlots.py

```diff
@@ -0,0 +1,361 @@
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+__all__ = ("BarPanel", "BarPlot")
+
+import operator as op
+from collections import defaultdict
+from typing import Mapping
+
+import matplotlib.pyplot as plt
+import numpy as np
+from lsst.analysis.tools.actions.plot.plotUtils import addPlotInfo
+from lsst.analysis.tools.interfaces import KeyedData, KeyedDataSchema, PlotAction, Vector
+from lsst.pex.config import Config, ConfigDictField, DictField, Field
+from matplotlib.figure import Figure
+from matplotlib.gridspec import GridSpec
+from matplotlib.patches import Rectangle
+
+
+class BarPanel(Config):
+    label = Field[str](
+        doc="Panel x-axis label.",
+        default="label",
+    )
+    bars = DictField[str, str](
+        doc="A dict specifying the bar graphs to be plotted in this panel. Keys are used to identify "
+        "bar graph IDs. Values are used to add to the legend label displayed in the upper corner of the "
+        "panel.",
+        optional=False,
+    )
+    yscale = Field[str](
+        doc="Y axis scaling.",
+        default="linear",
+    )
+
+
+class BarPlot(PlotAction):
+    panels = ConfigDictField(
+        doc="A configurable dict describing the panels to be plotted, and the bar graphs for each panel.",
+        keytype=str,
+        itemtype=BarPanel,
+        default={},
+    )
+    cmap = Field[str](
+        doc="Color map used for bar lines. All types available via `plt.cm` may be used. "
+        "A number of custom color maps are also defined: `newtab10`, `bright`, `vibrant`.",
+        default="newtab10",
+    )
+
+    def getInputSchema(self) -> KeyedDataSchema:
+        for panel in self.panels:  # type: ignore
+            for barData in self.panels[panel].bars.items():  # type: ignore
+                yield barData, Vector
+
+    def __call__(self, data: KeyedData, **kwargs) -> Mapping[str, Figure] | Figure:
+        return self.makePlot(data, **kwargs)
+        # table is a dict that needs: x, y, run, skymap, filter, tract,
+
+    def makePlot(
+        self, data: KeyedData, plotInfo: Mapping[str, str] = None, **kwargs  # type: ignore
+    ) -> Figure:
+        """Make an N-panel plot with a user-configurable number of bar graphs
+        displayed in each panel.
+
+
+        Parameters
+        ----------
+        data : `pandas.core.frame.DataFrame`
```

Comment:
data doesn't need to be a dataframe it should be able to take any form of keyed data.